### PR TITLE
ignore pip version check when offline 

### PR DESCRIPTION
**Pull request by lee212**
#591 pip version check is disabled when ConnectionError is raised.

### CODE REVIEW

**Review by andre-merzky — APPROVED** [review]
I agree this solves the problem.  However, maybe it would be better to silently ignore the condition?  It is not really an error to use RE offline, and I don't think it warrants a warning.  Either way, I leave this to your judgement.

**Comment by andre-merzky** [comment]
@lee212 : please note that the linters fail - please fix before merging.  Also, several lines are too long (we have an 80 char limit).

**Comment by lee212** [comment]
> I agree this solves the problem. However, maybe it would be better to silently ignore the condition? It is not really an error to use RE offline, and I don't think it warrants a warning. Either way, I leave this to your judgement.

I eliminated the warning at  236769ee23eb11dd867ef47b0c9ec07af16dc06c

**Comment by lee212** [comment]
> @lee212 : please note that the linters fail - please fix before merging. Also, several lines are too long (we have an 80 char limit).

Running linter on PR diff is failing because of `E0401: Unable to import`, for example,` 'hypothesis' (import-error)`
The current dependency is here: https://github.com/radical-cybertools/radical.entk/blob/devel/.github/workflows/python-app.yml#L98-L104, do we want to add all packages under pylint actions? Otherwise, I merge this PR for now and close.

**Comment by lee212** [comment]
I will merge this for now. I will fix pylint test if it complains again.

**Pull request merged**
